txmongo/_gridfs/__init__.py

```diff
@@ -62,6 +62,9 @@ def __init__(self, database, collection="fs"):
         self.__indexes_created_defer = defer.DeferredList([
             self.__files.create_index(
                 filter.sort(ASCENDING("filesname") + ASCENDING("uploadDate"))),
+            self.__files.create_index(
+                filter.sort(ASCENDING("filesname") + DESCENDING("uploadDate"))),
```

Comment:
After careful reading I've noticed that second index is redundant. `{filename: 1, uploadDate: 1}` index will cover both `find({filename: ...}).sort({uploadDate: 1})` and `find({filename: ...}).sort({uploadDate: -1})`.